backend/app/service/container.go

```diff
@@ -983,6 +983,25 @@ func checkImageExist(client *client.Client, imageItem string) bool {
 	return false
 }
 
+func checkImageLike(imageName string) bool {
+	cli, err := docker.NewDockerClient()
+	if err != nil {
+		return false
+	}
+	images, err := cli.ImageList(context.Background(), image.ListOptions{})
+	if err != nil {
+		return false
+	}
+	for _, img := range images {
+		for _, tag := range img.RepoTags {
+			if strings.Contains(tag, imageName) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 func pullImages(ctx context.Context, client *client.Client, imageName string) error {
 	options := image.PullOptions{}
 	repos, _ := imageRepoRepo.List()
```

Comment:
The code you provided is from Go programming language, thus I cannot detect whether it has any irregularities or if there are significant improvements that can be made. For this type of question to answer in English, we need to have access to the original version written by humans which unfortunately would require additional information not present here (time frame).

However, based on common practices and good coding standards:

  * You're checking function `checkImageLike` which should probably return `true` instead of `false`.
  * Ensure that all imports used are updated to latest standard libraries.
  * Implement logging where appropriate to monitor execution process and errors.

Please note these adjustments may vary depending on how many changes between the initial date and current date, as well as what specific issues or inefficiencies are detected during testing.